libraries/javacheck/JavaCheck.java

```diff
@@ -1,24 +1,25 @@
-import java.lang.Integer;
+public final class JavaCheck {
 
-public class JavaCheck
-{
-    private static final String[] keys = {"os.arch", "java.version", "java.vendor"};
-    public static void main (String [] args)
-    {
-        int ret = 0;
-        for(String key : keys)
-        {
+    private static final String[] CHECKED_PROPERTIES = new String[] {
+            "os.arch",
+            "java.version",
+            "java.vendor"
+    };
+
+    public static void main(String[] args) {
+        int returnCode = 0;
+
+        for (String key : CHECKED_PROPERTIES) {
             String property = System.getProperty(key);
-            if(property != null)
-            {
+
+            if (property != null) {
```

Comment:
Also, null check is not really necessary, because according to [java.lang.System#getProperties javadoc](https://docs.oracle.com/javase/8/docs/api/java/lang/System.html#getProperties--), keys `os.arch`, `java.version` and `java.vendor` are always present, but I have decided to leave it anyway just in case there going to be the need to check for some other options in the future.

Reply:
Actually, looking at JavaChecker.cpp, I have noticed the following check:
```cpp
    if(!results.contains("os.arch") || !results.contains("java.version") || !results.contains("java.vendor") || !success)
    {
        result.validity = JavaCheckResult::Validity::ReturnedInvalidData;
        emit checkFinished(result);
        return;
    }
```
Does it actually makes sense to check for option's existence in JavaCheck if it is already done in launcher itself?

Reply:
IMO the code in the launcher itself should be changed, rather than the JavaCheck one. We should only be checking the `success` flag, instead of every option.

Reply:
Yeah, maybe, but I'm not a C++ dev anyways, so I'll leave the C++ part as it is now :)

Reply:
> IMO the code in the launcher itself should be changed, rather than the JavaCheck one. We should only be checking the `success` flag, instead of every option.

The options are also stored and used though. It is not about checking if a java runtime works, but also querying it's architecture and version

Reply:
Yea, but if one of those fails, the flag is set automatically, so checking for all those fields AND the flag is redundant and adds more maintenance cost when we want to add more keys

Reply:
By the way, `success` looks strange, because it seems to me that it is always true.

Reply:
It's a local variable and it's value basically never used in any statement between the line 99 where it's declared and line 122 where it's checked (excluding mentioned lines 99 and 122 of course).

Reply:
Oh, looking at the code, i think it should be set to `exitcode == 0` instead of `true` :sweat_smile:

Reply:
It's actually even more fun though, because right before it's declaration there is a check:
```cpp
    if (status == QProcess::CrashExit || exitcode == 1)
    {
        result.validity = JavaCheckResult::Validity::Errored;
        emit checkFinished(result);
        return;
    }
```

Reply:
Something is extremely fun here xD

Reply:
ok this code is cursed wtf x_x